### ISSUE TITLE
Make progress bar optional

### ISSUE THREAD

**Issue by Rob--W** [mozilla/sign-addon#710]
See https://github.com/mozilla/web-ext/issues/2154 for context

The progress bar, initialized at https://github.com/mozilla/sign-addon/blob/af0fe2d2d6f2ac8321f7db0145db3b5495b8c66b/src/amo-client.js#L183-L188 (and implemented at https://github.com/mozilla/sign-addon/blob/af0fe2d2d6f2ac8321f7db0145db3b5495b8c66b/src/PseudoProgress.js) should be optional.

The `Client` is initialized by the `signAddon` function at https://github.com/mozilla/sign-addon/blob/af0fe2d2d6f2ac8321f7db0145db3b5495b8c66b/src/index.js#L95-L105
which is called by `web-ext sign` at https://github.com/mozilla/web-ext/blob/9c53dd4e4afc4be7111cc009a15dff8dea22f09d/src/cmd/sign.js#L115-L127

**Comment by NehaK745** [comment]
I would like to work on this issue. can you guide me on how to start?

**Comment by caitmuenster** [comment]
Welcome, @NehaK745! 👋

@willdurand, would you be able to provide some guidance about how to get started with this bug?

**Comment by willdurand** [comment]
> @willdurand, would you be able to provide some guidance about how to get started with this bug?

Hey, sorry. YES!

So there are two things:

A. what exactly we should do to fix this issue
B. how we can test `sign-addon` with AMO -dev (which will be required to test the changes that are required here)

Let's start with (B) first. One could follow these steps:

1. clone sign-addon, install the dependencies and build the library (see also: https://github.com/mozilla/sign-addon#development)
2. create an account on AMO -dev: https://addons-dev.allizom.org/
3. get API keys: https://addons-dev.allizom.org/en-US/developers/addon/api/key/
4. create a `run-sign-addon.js` file with the following content:

```js
var { signAddon } = require('./dist/sign-addon.js');

signAddon({
  xpiPath: '<ZIP FILE>',
  version: '1.0',
  apiKey: '<API KEY>',
  apiSecret: '<API SECRET>',
  apiUrlPrefix: 'https://addons-dev.allizom.org/api/v4',
  quiet: true,
})
  .then(function (result) {
    if (result.success) {
      console.log('The following signed files were downloaded:');
      console.log(result.downloadedFiles);
      console.log('Your extension ID is:');
      console.log(result.id);
    } else {
      console.error('Your add-on could not be signed!');
      console.error('Error code: ' + result.errorCode);
      console.error('Details: ' + result.errorDetails);
    }
    console.log(result.success ? 'SUCCESS' : 'FAIL');
  })
  .catch(function (error) {
    console.error('Signing error:', error);
  });
```

5. The actual API key and secret should be added to the file above (replacing `<API KEY>` and `<API SECRET>`)
6. Generate a dummy XPI on https://staticfil.es/webext-generator/, save it next to `run-sign-addon.js`, and replace `<ZIP FILE>` with the actual filename

After that, running `node run-sign-addon.js` should show the following output:

![Screenshot 2021-05-06 at 11 46 52](https://user-images.githubusercontent.com/217628/117278114-c0d1cb00-ae60-11eb-8e28-4fca5f71a278.png)

As we can see in the screenshot above, there is a row with some dots (and it is actually animated), which we can also see below:

![Screenshot 2021-05-06 at 11 49 52](https://user-images.githubusercontent.com/217628/117278585-2de56080-ae61-11eb-9013-79f380ed01a3.png)

This is the "(pseudo) progress bar" and that's what we would like to hide when a (new) option is passed to `sign-addon`. This should answer (A) above.

Let me know if you have more specific questions.

**Comment by manthony9** [comment]
@willdurand I am trying to work on this issue and can't seem to get myself authenticated following the steps you mentioned.
![image](https://user-images.githubusercontent.com/13238071/117588168-c1cc5c00-b0ef-11eb-84ea-9c0a4ad23f63.png)

Am i calling the xpi path correctly ?

**Comment by manthony9** [comment]
@willdurand yup. I even tried v5 vs v4 but still got the same message. Could it be something to do with the dummyXPI ?
![image](https://user-images.githubusercontent.com/13238071/117712272-1a5c3180-b1a2-11eb-8386-4c69cb54509f.png)
![image](https://user-images.githubusercontent.com/13238071/117712429-5099b100-b1a2-11eb-8d75-d4f74451bfeb.png)

**Comment by willdurand** [comment]
@manthony9 look at the url on your screenshot and the one I pasted above. They are different. Your screenshot shows the production instance, not dev.

**Comment by manthony9** [comment]
Oh! that's correct. Thank you @willdurand.  I think I ended up at the production instance cause my dev instance is permanently stuck on this page. Any way I can get around this?
![image](https://user-images.githubusercontent.com/13238071/117721613-146c4d80-b1ae-11eb-8135-2a436dd11f15.png)

**Comment by willdurand** [comment]
> Oh! that's correct. Thank you @willdurand. I think I ended up at the production instance cause my dev instance is permanently stuck on this page. Any way I can get around this? Haven't received any email in spam or trash.
> ![image](https://user-images.githubusercontent.com/13238071/117721613-146c4d80-b1ae-11eb-8135-2a436dd11f15.png)

ah wait. let me find the email and forward it to you (we don't automatically send emails on -dev)

**Comment by willdurand** [comment]
@manthony9 did you delete your dev account?

**Comment by manthony9** [comment]
Nope , it should be there with the email-id max.payne940@gmail.com

**Comment by manthony9** [comment]
@willdurand is this close to the end-result we want? 
![image](https://user-images.githubusercontent.com/13238071/117855989-12ab9400-b259-11eb-8204-b6c42c6e9beb.png)

**Comment by willdurand** [comment]
I think so

**Comment by manthony9** [comment]
Hi @willdurand @Rob--W  just submitted my pull request. Still learning about open-source contributions so please feel free to let me know if there's something wrong.

**Comment by Rob--W** [comment]
@manthony9 Thanks for your PR (pull request). I will defer to Will for the code review since he's the mentor of this issue. I do have two points of feedback (not specific to the code):

When you submit a PR on Github, also put the URL of the issue that you're fixing in the bug, along with a special keyword. Then Github will automatically close the issue when the PR is merged. For example: `Fixes #710.` - as explained at https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

The commits that you have created seem to not be recognized by Github as belonging to your account. Usually, Github will turn the name in a link to the Github profile that authored the commit. Your commit doesn't have a name with a link and the avatar is blank. This often happens when the email is incorrectly configured on Git. When I check out your patch locally (as shown below), I do see a valid-looking email. So in your case I guess that your Git is configured with an email address that differs from the one of your Github account. Either edit the commit to use the e-mail address that you use for Github, or register that email address with your Github profile, by following the instructions at https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/adding-an-email-address-to-your-github-account

```
$ git clone https://github.com/mozilla/sign-addon
$ cd sign-addon
$ git fetch origin refs/pull/755/head
$ git log FETCH_HEAD
```

**Comment by manthony9** [comment]
@Rob--W thank you for the feedback. That is strange, I just added the only other email-id I use onto my profile, assuming that should solve the issue. I forked the mozilla repo, changed the origin to my forked repo, and pushed my commits that way, not sure if that is the right approach.
